compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -1600,57 +1603,93 @@ object SymDenotations {
        *    and this changes subtyping relations. As a shortcut, we do not
        *    cache ErasedValueType at all.
        */
-      def isCachable(tp: Type, btrCache: BaseTypeRefMap): Boolean = {
+      def isCachable(tp: Type, btrCache: BaseTypeMap): Boolean = {
         def inCache(tp: Type) = btrCache.containsKey(tp)
         tp match {
           case _: TypeErasure.ErasedValueType => false
           case tp: TypeRef if tp.symbol.isClass => true
           case tp: TypeVar => tp.inst.exists && inCache(tp.inst)
-          case tp: TypeProxy => inCache(tp.underlying)
-          case tp: AndOrType => inCache(tp.tp1) && inCache(tp.tp2)
+          //case tp: TypeProxy => inCache(tp.underlying) // disabled, can re-enable insyead of last two lines for performance testing
+          //case tp: AndOrType => inCache(tp.tp1) && inCache(tp.tp2)
+          case tp: TypeProxy => isCachable(tp.underlying, btrCache)
```

Comment:
It seems we were needlessly pessimistic before and therefore cached less than we could have.